storagenode/console/consoleserver/server.go

```diff
@@ -79,166 +102,230 @@ func NewServer(logger *zap.Logger, config Config, service *console.Service, list
+	if satelliteID != nil {
+		return server.service.GetBandwidthBySatellite(ctx, *satelliteID)
+	}
+
+	return server.service.GetUsedBandwidthTotal(ctx)
+}
+
+func (server *Server) getBandwidthChartData(ctx context.Context, satelliteID *storj.NodeID) (_ []console.BandwidthUsed, err error) {
+	from, to := getMonthRange()
+
+	if satelliteID != nil {
+		return server.service.GetDailyBandwidthUsed(ctx, *satelliteID, from, to)
+	}
 
-	w.WriteHeader(http.StatusOK)
+	return server.service.GetDailyTotalBandwidthUsed(ctx, from, to)
 }
 
-func (s *Server) getBandwidth(ctx context.Context, satelliteID *storj.NodeID) (_ *console.BandwidthInfo, err error) {
+func (server *Server) getDiskSpaceChartData(ctx context.Context, satelliteID *storj.NodeID, satellitesID storj.NodeIDList) (result []nodestats.SpaceUsageStamp, err error) {
+	from, to := getMonthRange()
+
 	if satelliteID != nil {
-		return s.service.GetBandwidthBySatellite(ctx, *satelliteID)
+		return server.service.GetDailyStorageUsedForSatellite(ctx, *satelliteID, from, to)
+	}
+
+	if satellitesID == nil || satellitesID.Len() == 0 {
+		return nil, nil
 	}
 
-	return s.service.GetUsedBandwidthTotal(ctx)
+	for _, id := range satellitesID {
+		usage, err := server.service.GetDailyStorageUsedForSatellite(ctx, id, from, to)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(result) == 0 {
+			result = usage
+		} else {
+			for i := 0; i < len(result); i++ {
+				for j := 0; j < len(usage); j++ {
+					if result[i].TimeStamp == usage[j].TimeStamp {
+						result[i].AtRestTotal += usage[j].AtRestTotal
+					}
+				}
+			}
+		}
+	}
+
+	return result, nil
 }
 
-func (s *Server) getStorage(ctx context.Context, satelliteID *storj.NodeID) (_ *console.DiskSpaceInfo, err error) {
+func (server *Server) getStorage(ctx context.Context, satelliteID *storj.NodeID) (_ *console.DiskSpaceInfo, err error) {
 	if satelliteID != nil {
-		return s.service.GetUsedStorageBySatellite(ctx, *satelliteID)
+		return server.service.GetUsedStorageBySatellite(ctx, *satelliteID)
 	}
 
-	return s.service.GetUsedStorageTotal(ctx)
+	return server.service.GetUsedStorageTotal(ctx)
 }
 
-func (s *Server) checkSatelliteID(satelliteIDs storj.NodeIDList, satelliteID storj.NodeID) error {
+func (server *Server) checkSatelliteID(satelliteIDs storj.NodeIDList, satelliteID storj.NodeID) error {
 	for _, id := range satelliteIDs {
 		if satelliteID == id {
 			return nil
 		}
 	}
 
-	return errs.New("satellite id in not found in the available satellite list")
+	return errs.New("satellite id is not found in the available satellite list")
 }
 
-func (s *Server) parseSatelliteIDParam(satelliteID string) (*storj.NodeID, error) {
+func (server *Server) parseSatelliteIDParam(satelliteID string) (*storj.NodeID, error) {
 	if satelliteID != "" {
 		id, err := storj.NodeIDFromString(satelliteID)
 		return &id, err
 	}
 
 	return nil, nil
 }
+
+// getMonthRange is used to get first and last dates of month
+func getMonthRange() (firstDay, lastDay time.Time) {
```

Comment:
NIT: I found this function in a lot of places, might be a good idea to make it a utils function.